lib/algorithms.js

```diff
@@ -0,0 +1,72 @@
+'use strict';
+
+const { AES_CTR, AES_CBC, AES_GCM, AES_KW } = require('./algorithms/aes');
+const { HKDF } = require('./algorithms/hkdf');
+const { PBKDF2 } = require('./algorithms/pbkdf2');
+const { SHA_1, SHA_256, SHA_384, SHA_512 } = require('./algorithms/sha');
+const { NotSupportedError } = require('./errors');
+
+const algorithms = [
+  AES_CTR,
+  AES_CBC,
+  AES_GCM,
+  AES_KW,
+
+  HKDF,
+
+  PBKDF2,
+
+  SHA_1,
+  SHA_256,
+  SHA_384,
+  SHA_512
+];
+
+function objectFromArray(array, fn) {
+  const obj = {};
+  for (const val of array)
+    fn(obj, val);
+  return obj;
+}
+
+const supportedAlgorithms = objectFromArray([
```

Comment:
This object, its name and its structure are specified in [18.4.2](https://www.w3.org/TR/WebCryptoAPI/#dfn-supportedAlgorithms) and [18.4.3](https://www.w3.org/TR/WebCryptoAPI/#algorithm-normalization-define-an-algorithm), I added a `Spec:` line to the code.

It also kind of makes sense in my opinion, since `supportedAlgorithms[op][algorithmName]` returns an algorithm, not an operation.